client/landing/gutenboarding/onboarding-block/vertical-select/index.tsx

```diff
@@ -0,0 +1,129 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { useState, useCallback, useRef } from 'react';
+import { __ as NO__ } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import Suggestions from '../../../../components/suggestions';
+import { STORE_KEY } from '../../store';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+import { SiteVertical, isFilledFormValue } from '../../store/types';
+
+interface Props {
+	inputClass: string;
+}
+
+export default function VerticalSelect( { inputClass }: Props ) {
+	const popular = [
+		NO__( 'Travel Agency' ),
+		NO__( 'Digital Marketing' ),
+		NO__( 'Cameras & Photography' ),
+		NO__( 'Website Designer' ),
+		NO__( 'Restaurants' ),
+		NO__( 'Fashion Designer' ),
+		NO__( 'Real Estate Agent' ),
+	];
+
+	const [ inputValue, setInputValue ] = useState( '' );
+	const [ suggestionsVisibility, setsuggestionsVisibility ] = useState( false );
+
+	const suggestionRef = useRef< Suggestions >( null );
+	const inputRef = useRef< HTMLInputElement >( null );
+
+	const verticals = useSelect( select =>
+		select( STORE_KEY )
+			.getVerticals()
+			.map( x => ( {
+				label: x.vertical_name,
+				id: x.vertical_id,
+			} ) )
+	);
```

Comment:
For intensive transforms like this, we should probably use a dedicated selector with memoization. I believe [`rememo`](https://www.npmjs.com/package/rememo) is what Gutenberg is using.

`useSelect` has some memoization built in I believe, but I suspect we may break memoization with this sort of `useSelect` code.